mesh/mesh.go

```diff
@@ -96,6 +101,27 @@ func NewMesh(db *MeshDB, atxDb AtxDB, rewardConfig Config, mesh MeshValidator, t
 	return ll
 }
 
+func NewRecoveredMesh(db *MeshDB, atxDb AtxDB, rewardConfig Config, mesh MeshValidator, txInvalidator TxMemPoolInValidator, atxInvalidator AtxMemPoolInValidator, pr TxProcessor, logger log.Log) *Mesh {
+	ll := NewMesh(db, atxDb, rewardConfig, mesh, txInvalidator, atxInvalidator, pr, logger)
+
+	latest, err := db.general.Get(LATEST)
+	if err != nil {
+		logger.Panic("could not recover latest layer")
+	}
+
+	ll.latestLayer = types.LayerID(util.BytesToUint64(latest))
+
+	validated, err := db.general.Get(VALIDATED)
```

Comment:
I think there could be a potential bug here, latest could be written and validated could not exist yet. which will cause a panic when loading from memory

Reply:
the outcome would just be revalidating the "latest" is the important issue worst case scenario we revalidate a layer, also for our needs now i think this is good enough

Reply:
if you have the `LATEST` flag set but no `VALIDATED` flag this will panic. this case can happen and in which the node will not be able to boot

Reply:
happens only if we didn't validate layer 1 you can just start fresh in that case

Reply:
How would we start fresh at this case? imagine a user running the node for one second and then shutting it down, `LATEST` will be written and therefore `PersistentData()` will be true, causing load from disk, after which the `VALIDATED` flag will not be found and this will cause this code to panic.
Just add a check if `VALIDATED` also exist in `PersistentData()` func

Reply:
only if he shuts down before he validates layer 1 which is very unlikely also i'm imagining us running tests using this not the hoe user just yet we have more importent things to fix before we make this feature airtight

Reply:
youre right, it's not a common bug, bit still can happen without possibility to recover unless you delete the DB files. It's just one line of code to fix it, I think that if we can make our code better with minimal effort we should.

Reply:
if it was one line i would fix it, i think we have more rare cases where we would have data corruption, there is a refactor to do for this to be better but it would take some effort and i don't think now is the right time, i can easily automatically override the existing data but i rather we save it and analyse what went wrong for now its manual later maybe we would do something smarter